auth/src/api/handlers.rs

```diff
@@ -75,6 +75,36 @@ pub(crate) async fn put_user_reset_key(
     user_manager.reset_key(account_name).await
 }
 
+pub(crate) async fn post_subscription(
+    _: Admin,
+    State(user_manager): State<UserManagerState>,
+    Path(account_name): Path<AccountName>,
+    payload: Json<SubscriptionRequest>,
+) -> Result<(), Error> {
+    user_manager
+        .insert_subscription(
+            &account_name,
+            &payload.id,
+            &payload.r#type,
+            payload.quantity,
+        )
+        .await?;
+
+    Ok(())
+}
+
+pub(crate) async fn delete_subscription(
```

Comment:
I am wondering about when this will be called. If a user cancels their RDS subscription, will it be cancelled immediately, or at the end of the period? If it's cancelled immediately from the console, it makes sense for the console to call this after. If it's at the end of the period, this would have to be called at the end of the period.

Reply:
Based on the TR I was thinking the console just calls this immediately for the time being

Reply:
Ah, okay! We should just make sure that the console also cancels the subscription immediately then, and if the user has already paid for future dates, they should be refunded. cc @GugaGongadze

Reply:
If we delete the subscription immediately we could also delete the RDS, but that's out of scope for this PR. And we might change to delete at period end in the future.